lib/runners/dart.js

```diff
@@ -0,0 +1,171 @@
+'use strict'
+
+const shovel = require('../shovel');
+const util = require('../util');
+const config = require('../config');
+const fs = require('fs');
+const path = require('path');
+const temp = require('temp').track();
+
+const dir = '/tmp/dart';
+
+module.exports.run = function run(opts, cb) {
+  shovel.start(opts, cb, {
+    solutionOnly: function(run) {
+      _createSolution(opts).then(file => {
+        // Run
+        run({
+          name: 'dart',
+          args: [file],
+          options: {
+            cwd: dir
+          }
+        });
+      });
+    },
+    testIntegration: function(run) {
+      // test is the only one supported right now...
+      if (opts.testFramework != 'test')
+        throw 'Test framework is not supported';
+
+      // Create the spec test
+      _createSpec(opts).then(file => {
+        // Run
+        run({
+          name: 'pub',
+          args: ['run', 'test', file],
+          options: {
+            cwd: dir
+          }
+        });
+      });
+    },
+    sanitizeStdOut: function(stdout) {
+      if (opts.fixture) return _format(stdout)
+      else return stdout;
+    }
+  });
+};
+
+// Create the solution file
+var _createSolution = function(opts) {
+  // Set the code, _block() will stop user adding imports
+  // that are not provided in setup
+  let code = `
+    ${opts.setup ? opts.setup : ''}
+    _block() {}
```

Comment:
This does actually stop users creating a function called `_block`... I think that the idea is required otherwise they can put any imports they want in their code if its not one with tests. Hmmmm... Maybe a more abstract name that no one would ever use?